driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -267,17 +295,23 @@ public void doMaintenance() {
         }
     }
 
-    private PooledConnection getPooledConnection(final long timeout, final TimeUnit timeUnit) {
-        UsageTrackingInternalConnection internalConnection = pool.get(timeout, timeUnit);
-        while (shouldPrune(internalConnection)) {
-            pool.release(internalConnection, true);
-            internalConnection = pool.get(timeout, timeUnit);
-        }
-        connectionPoolListener.connectionCheckedOut(new ConnectionCheckedOutEvent(internalConnection.getDescription().getConnectionId()));
-        if (LOGGER.isTraceEnabled()) {
-            LOGGER.trace(format("Checked out connection [%s] to server %s", getId(internalConnection), serverId.getAddress()));
+    private PooledConnection getPooledConnection(final long timeout, final TimeUnit timeUnit, final boolean beginCheckout) {
+        try {
+            if (beginCheckout) {
+                connectionPoolListener.connectionCheckOutStarted(new ConnectionCheckOutStartedEvent(serverId));
+            }
+            UsageTrackingInternalConnection internalConnection = pool.get(timeout, timeUnit);
+            while (shouldPrune(internalConnection)) {
+                pool.release(internalConnection, true);
+                internalConnection = pool.get(timeout, timeUnit);
+            }
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace(format("Checked out connection [%s] to server %s", getId(internalConnection), serverId.getAddress()));
+            }
+            return new PooledConnection(internalConnection);
+        } catch (Exception e) {
```

Comment:
The try catch block is not necessary if all the catch block is doing is re-throwing the unaltered exception.